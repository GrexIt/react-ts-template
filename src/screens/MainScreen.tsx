import * as React from 'react';
import { ThemeContextProvider } from '../contexts/ThemeContext';
import { ChangeTheme } from './components/ChangeTheme';
import { Hello } from './components/Hello';

export const Main = () => {
  console.log('rendering main');
  return (
    <ThemeContextProvider>
      <Hello />
      <ChangeTheme />
    </ThemeContextProvider>
  );
};
