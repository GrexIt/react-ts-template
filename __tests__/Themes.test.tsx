'use strict';

import * as Enzyme from 'enzyme';
import { mount } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Main } from '../src/screens/MainScreen';

// Enzyme.configure({ adapter: new Adapter() });
Enzyme.configure({ adapter: new Adapter() });

it('default light theme', () => {
  const wrapper = mount(<Main />);
  expect(wrapper.html()).toMatchSnapshot();
});

it('change to dark theme should work', () => {
  const wrapper = mount(<Main />);
  wrapper.find('button').simulate('click');
  expect(wrapper.html()).toMatchSnapshot();
});
