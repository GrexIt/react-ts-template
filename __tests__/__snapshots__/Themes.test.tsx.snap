// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  style={
    Object {
      "backgroundColor": "white",
      "fontSize": 45,
    }
  }
>
  Hello from MainScreen!!
</p>
`;
